🛠 Tooling: Release job blocked from pushing to GitHub

### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/ts-api-utils/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

This GitHub Action `release.yml` Release job failed to push to GitHub: https://github.com/JoshuaKGoldberg/ts-api-utils/actions/runs/4115398512/jobs/7104053426

```plaintext
 M package.json
ERROR remote: Permission to JoshuaKGoldberg/ts-api-utils.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/JoshuaKGoldberg/ts-api-utils/': The requested URL returned error: 403
```

A later run that fails to publish a package version to npm twice (since the Git commit that bumped version never was pushed to GitHub): https://github.com/JoshuaKGoldberg/ts-api-utils/actions/runs/4115932211/jobs/7105309725

```plaintext
npm notice Publishing to https://registry.npmjs.org/
npm ERR! code E403
npm ERR! 403 403 Forbidden - PUT https://registry.npmjs.org/ts-api-tools - You cannot publish over the previously published versions: 0.0.18.
npm ERR! 403 In most cases, you or one of your dependencies are requesting
npm ERR! 403 a package version that is forbidden by your security policy, or
npm ERR! 403 on a server you do not have access to.
```

...but I set up a dev PAT token. Why is this broken?

### Additional Info

_No response_

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -5,6 +5,9 @@ on:
     branches:
       - main
 
+permissions:
+  contents: write
+
 jobs:
   release:
     runs-on: ubuntu-latest
@@ -34,7 +37,7 @@ jobs:
               }
             }
       - env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_TOKEN: ${{ github.token }}
         run: if pnpm run should-semantic-release ; then pnpm release-it --verbose ; fi
       - if: always()
         name: Recreate branch protection on main

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
 	"name": "ts-api-utils",
-	"version": "0.0.22",
+	"version": "0.0.23",
 	"description": "Utility functions for working with TypeScript's API. Successor to the wonderful tsutils.",
 	"repository": {
 		"type": "git",

